### PR TITLE
Add ability to get at the current QApplication

### PR DESCRIPTION
**Pull request by tpickett66**
Providing access to this function eliminates the need for language
specific bindings to be forced to re-implement tracking of this object.

### CODE REVIEW

**Comment by seanchas116** [comment]
Thank you!
Can you add a header declaration and a test for this function?

**Comment by tpickett66** [comment]
Done, sorry to have forgotten that.

**Comment by seanchas116** [comment]
Thanks for fixing Travis build!

Sorry for not clarifying, can you add a test case in [application_test.cpp](https://github.com/seanchas116/libqmlbind/blob/master/test/application_test.cpp)?

Like: 

```c++
SECTION("instance")
{
    REQUIRE(qmlbind_application_instance() != nullptr);
}
```

File: qmlbind/src/api_application.cpp [hunk 1]
```diff
@@ -91,6 +91,11 @@ qmlbind_application qmlbind_application_new(int argc, char **argv)
     return app;
 }
 
+qmlbind_application qmlbind_application_instance(void)
+{
+    return static_cast <QApplication*> (QCoreApplication::instance());
```
Comment (seanchas116) [inline qmlbind/src/api_application.cpp]: How about using `http://doc.qt.io/qt-5/qapplication.html#qApp` macro?
It doesn't need static cast.

**Comment by tpickett66** [comment]
Doh, I didn't fully read your request for the header *and* test; I'll add that straight away. :-)

File: qmlbind/src/api_application.cpp (hunk 1 above)
Comment (tpickett66) [inline qmlbind/src/api_application.cpp]: Yeah, I can make that change, the documentation I was looking at only mentioned the `instance()` static method.

**Comment by seanchas116** [comment]
Looks good. Thanks for pull request!

**Pull request merged**